lib/services/local/access_list.go

```diff
@@ -57,44 +60,109 @@ func NewAccessListService(backend backend.Backend, clock clockwork.Clock) (*Acce
 		return nil, trace.Wrap(err)
 	}
 
+	almSvc, err := generic.NewService(&generic.ServiceConfig[*accesslist.AccessListMember]{
+		Backend:       backend,
+		PageLimit:     accessListMemberMaxPageSize,
+		ResourceKind:  types.KindAccessListMember,
+		BackendPrefix: accessListMemberPrefix,
+		MarshalFunc:   services.MarshalAccessListMember,
+		UnmarshalFunc: services.UnmarshalAccessListMember,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
 	return &AccessListService{
-		log:   logrus.WithFields(logrus.Fields{trace.Component: "access-list:local-service"}),
-		clock: clock,
-		svc:   svc,
+		log:    logrus.WithFields(logrus.Fields{trace.Component: "access-list:local-service"}),
+		clock:  clock,
+		alSvc:  alSvc,
+		almSvc: almSvc,
 	}, nil
 }
 
 // GetAccessLists returns a list of all access lists.
 func (a *AccessListService) GetAccessLists(ctx context.Context) ([]*accesslist.AccessList, error) {
-	accessLists, err := a.svc.GetResources(ctx)
+	accessLists, err := a.alSvc.GetResources(ctx)
 	return accessLists, trace.Wrap(err)
 }
 
 // ListAccessLists returns a paginated list of access lists.
 func (a *AccessListService) ListAccessLists(ctx context.Context, pageSize int, nextToken string) ([]*accesslist.AccessList, string, error) {
-	return a.svc.ListResources(ctx, pageSize, nextToken)
+	return a.alSvc.ListResources(ctx, pageSize, nextToken)
 }
 
 // GetAccessList returns the specified access list resource.
 func (a *AccessListService) GetAccessList(ctx context.Context, name string) (*accesslist.AccessList, error) {
-	accessList, err := a.svc.GetResource(ctx, name)
+	accessList, err := a.alSvc.GetResource(ctx, name)
 	return accessList, trace.Wrap(err)
 }
 
 // UpsertAccessList creates or updates an access list resource.
 func (a *AccessListService) UpsertAccessList(ctx context.Context, accessList *accesslist.AccessList) (*accesslist.AccessList, error) {
-	if err := trace.Wrap(a.svc.UpsertResource(ctx, accessList)); err != nil {
+	if err := trace.Wrap(a.alSvc.UpsertResource(ctx, accessList)); err != nil {
 		return nil, trace.Wrap(err)
 	}
 	return accessList, nil
 }
 
 // DeleteAccessList removes the specified access list resource.
 func (a *AccessListService) DeleteAccessList(ctx context.Context, name string) error {
-	return trace.Wrap(a.svc.DeleteResource(ctx, name))
+	err := a.alSvc.DeleteResource(ctx, name)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	// Delete all associated members as well.
+	return trace.Wrap(a.DeleteAllAccessListMembersForAccessList(ctx, name))
 }
 
 // DeleteAllAccessLists removes all access lists.
 func (a *AccessListService) DeleteAllAccessLists(ctx context.Context) error {
-	return trace.Wrap(a.svc.DeleteAllResources(ctx))
+	err := trace.Wrap(a.alSvc.DeleteAllResources(ctx))
+	if err != nil {
+		return trace.Wrap(err)
+	}
```

Comment:
Why double-wrapping?

```suggestion
	if err := a.alSvc.DeleteAllResources(ctx); err != nil {
		return trace.Wrap(err)
	}
```

Reply:
Bad copy-paste! Fixed.